src/coreclr/jit/lclvars.cpp

```diff
@@ -650,13 +651,23 @@ void Compiler::lvaInitUserArgs(InitVarDscInfo* varDscInfo, unsigned skipArgs, un
         CorInfoTypeWithMod corInfoType = info.compCompHnd->getArgType(&info.compMethodInfo->args, argLst, &typeHnd);
         varDsc->lvIsParam              = 1;
 
-        lvaInitVarDsc(varDsc, varDscInfo->varNum, strip(corInfoType), typeHnd, argLst, &info.compMethodInfo->args);
+        CorInfoType type = strip(corInfoType);
+        lvaInitVarDsc(varDsc, varDscInfo->varNum, type, typeHnd, argLst, &info.compMethodInfo->args);
 
-        if (strip(corInfoType) == CORINFO_TYPE_CLASS)
+        if (type == CORINFO_TYPE_CLASS)
         {
             CORINFO_CLASS_HANDLE clsHnd = info.compCompHnd->getArgClass(&info.compMethodInfo->args, argLst);
             lvaSetClass(varDscInfo->varNum, clsHnd);
         }
+        else if (type == CORINFO_TYPE_BYREF)
```

Comment:
Do you mean something like this:
```cs
unsafe void BlendVariable(string* a, string b)
{
    *a = b;
}
```
isn't not safe to remove WB here and, hopefully, nobody does this anyway 🙂

Reply:
Yes, it would be unsafe to remove the write barrier there.

I think it means we should make this feature work for `CORINFO_TYPE_PTR` too, for parity between managed and unmanaged pointers. E.g.
```
    [MethodImpl(MethodImplOptions.NoInlining)]
    void Test(MyStruct* s, object o1, object o2)
    {
        s->A = o1;
        s->B = o2;
    }

    ref struct MyStruct
    {
        public object A;
        public object B;
    }
```
should avoid the write barrier as well, like if it was a managed pointer.

Reply:
NOTE that `MyStruct* s` is issueing a warning in Roslyn as an attempt to take an address of a managed type. Do we want to also eagerly load all types under unmanaged pointers in hope somebody do this (I presume it's rare/never).

cc @jkotas

Reply:
I think it is fine to assume that both `ref MyByRefLikeStruct` and `MyByRefLikeStruct*` never point into GC heap

Reply:
What I'm after is that we have consistency in optimizations like these. Switching a byref to a pointer because the target is on the stack should not come with surprising perf regressions like new write barriers.